### ISSUE TITLE
ActiveStorage direct_upload failing in IE11 - Assignment to read-only properties is not allowed in strict mode

### ISSUE THREAD

**Issue by kylekeesling** [rails/rails#32378]
## Steps to reproduce
Using Active Storage, along with direct upload, upload a file directly to a service (tested with S3) using Internet Explorer 11.

## Expected behavior
The file should be properly uploaded via javascript directly to the service.

## Actual behavior
IE throws an error when the file upload is complete to S3 and the form submits:
```
Assignment to read-only properties is not allowed in strict mode
File: activestorage.self-*.js, Line: 1, Column: 3274
```
Which I tracked down to this line of code:

https://github.com/rails/rails/blob/f9ec3e5cc3753b7a46cc1a34d57ca941e4cf7211/activestorage/app/javascript/activestorage/ujs.js#L62

I'm still wrapping my head around this code but it seems to me as though the form is being modified after it's submitted, and IE doesn't like it. I'm happy to help correct it if someone can point me in the direction.

**System configuration**
Rails version:
5.2.0rc2

**Ruby version**:
ruby 2.5.0

**Comment by robban** [comment]
You can try changing the row to:
`button.style.cssText = "display:none"`

**Comment by AlexKovynev** [comment]
Or maybe button.style.display = "none"

**Comment by kylekeesling** [comment]
I think I've figured it out.

There is a minified javascript file (activestorage/app/assets/javascripts/activestorage.js) that seems to be taking precedence over the code in app/javascript.

When I removed that code everything seems to work properly, also, I stylistically updated the code in ujs.js to:

`button.style.display = "none"`

I'll submit a pull request of my updates shortly.

**Comment by kylekeesling** [comment]
Actually, nevermind. I believe the issue is just that the asset pipeline version needs to be updated/fixed.

When I install and use ActiveStorage through webpack everything works fine.

**Comment by kylekeesling** [comment]
After a few false starts, and finding some hints on how to rebuild the webpack assets for the asset pipeline (see #31860), I was able to do some more testing, and it looks as though the problem is in this area, but rather with the setting of the input type.

I switched from:

`button.type = "submit"`

to

`button.setAttribute("type", "submit")`

And everything works great.

The problem that I'm still seeing is that if the form is set to `remote=true` the form submission does not seem to wait for the direct upload to complete before it's submitted. I'm not sure if `direct_upload` is designed to be used with remote forms at this point, but thought it was worth noting.

I'll submit my pull request shortly.
